### DISCUSSION TITLE
Confusion with Version Changes: Difficulty Handling API Changes for New Bevy Users

### DISCUSSION THREAD

**Discussion by brantcass** [bevyengine/bevy#16429] [Q&A]
As a new Bevy user, I’ve encountered significant confusion related to changes in the API, particularly with methods like add_startup_system() and add_startup_systems(). After replacing the old add_startup_systems() with the new add_startup_system() in my code, I ran into the following error:

app.add_startup_system(setup_terrain_system);

no method named `add_startup_system` found for mutable reference `&mut bevy::prelude::App` in the current scope
While researching the issue, I saw that other people have experienced this issue, especially regarding version mismatches and the deprecation of certain methods. As a beginner, trying to navigate these changes could be really challenging, and it has taken me a considerable amount of time to understand what might be happening. However, I’m still not entirely sure how to proceed, and the lack of clear guidance in the documentation for newer users has made it harder to resolve this.

I believe this issue might be related to the version changes, but as someone new to Bevy, understanding how to handle these differences is not immediately clear. It would be really helpful if the documentation could provide more specific guidance for beginners on dealing with API changes between versions. After looking at the documentation I found a few parts that brought it up but nothing that helped me figure out this issue. I could be completely in the wrong but this is what I have experienced so far.

**Comment by torsteingrindvik** [comment]
You're right, there is a lot of churn.
The change you're describing was a while back.
There are migration guides for affected APIs on new releases, here's the one you're affected by: https://bevyengine.org/learn/migration-guides/0-10-to-0-11/#schedule-first-the-new-and-improved-add-systems

**Reply by yilisharcs** [reply]
Is it possible to note down these migration changes in the compiler error messages?
